---
int64 encoder_counts